### PR TITLE
Add pre-approved supressions to fix lint diff gate failures on Oct 24 Preview

### PR DESCRIPTION
**Pull request by ragovada**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [ ] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [x] Other, please clarify:
    - This PR has fixes for the lint diff failures on its parent PR https://github.com/Azure/azure-rest-api-specs/pull/30112. All the suppressions are pre-approved in the previous releases.

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

File: specification/machinelearningservices/resource-manager/readme.md [hunk 1]
```diff
@@ -41,6 +41,45 @@ input-file:
   - Microsoft.MachineLearningServices/preview/2024-10-01-preview/registries.json
   - Microsoft.MachineLearningServices/preview/2024-10-01-preview/workspaceFeatures.json
   - Microsoft.MachineLearningServices/preview/2024-10-01-preview/workspaceRP.json
+suppressions:
+  -code: AvoidAdditionalProperties
+   reason: As discussed these are hyperparameters which can vary by model and fine tuning task types so cannot have 
+           strictly typed properties.
```
Comment (ZhidaLiu) [inline specification/machinelearningservices/resource-manager/readme.md:47]: spacing here is wrong?

File: specification/machinelearningservices/resource-manager/readme.md [hunk 2]
```diff
@@ -41,6 +41,45 @@ input-file:
...
+   where:
+      - $.definitions["CustomModelFineTuning"].properties["hyperParameters"]
+  -code: AvoidAdditionalProperties
+   reason: These schemas are already in production use,.
```
Comment (ZhidaLiu) [inline specification/machinelearningservices/resource-manager/readme.md:51]: extra ","

File: specification/machinelearningservices/resource-manager/readme.md [hunk 3]
```diff
@@ -41,6 +41,45 @@ input-file:
...
+   where:
+      - $.definitions.CustomKeysWorkspaceConnectionProperties.properties.credentials.properties.keys
+      - $.definitions.CustomKeys.properties.keys
+  - code: DefinitionsPropertiesNamesCamelCase
+    reason: CMK is a short term and not violate the camel case rule.
+    where:
+      - $.definitions.WorkspaceProperties.properties.enableServiceSideCMKEncryption
+  - code: GuidUsage
+    reason: This property has always been a GUID, we just didn't mark its format before, this can't be change without 
+            breaking the customer.
```
Comment (ZhidaLiu) [inline specification/machinelearningservices/resource-manager/readme.md:61]: spacing is wrong?

**Comment by ragovada** [comment]
FYI ARM Reviewer: The goal of this PR is to fix the lint diff failures on this parent PR : https://github.com/Azure/azure-rest-api-specs/pull/30112. Also, all the suppressions added here were pre-approved by ARM in the previous releases. You can verify the same here: https://github.com/ragovada/azure-rest-api-specs/blob/main/specification/machinelearningservices/resource-manager/readme.md

**Comment by razvanbadea-msft** [comment]
> FYI ARM Reviewer: The goal of this PR is to fix the lint diff failures on this parent PR : #30112. Also, all the suppressions added here were pre-approved by ARM in the previous releases. You can verify the same here: https://github.com/ragovada/azure-rest-api-specs/blob/main/specification/machinelearningservices/resource-manager/readme.md

why are you doing it in a separate PR ?

**Comment by ragovada** [comment]
> > FYI ARM Reviewer: The goal of this PR is to fix the lint diff failures on this parent PR : #30112. Also, all the suppressions added here were pre-approved by ARM in the previous releases. You can verify the same here: https://github.com/ragovada/azure-rest-api-specs/blob/main/specification/machinelearningservices/resource-manager/readme.md
> 
> why are you doing it in a separate PR ?

Hi @razvanbadea-msft - We made a separate PR because we cannot modify anything directly on the release branch. The release * branches are protected as mentioned here : https://eng.ms/docs/products/azure-developer-experience/design/api-specs-pr/api-versions-and-branches and we have been doing this historically.

**Review by ZhidaLiu — APPROVED** [review]

**Comment by ragovada** [comment]
@ms-bogdan-rotaru - Could you please approve the suppression as well? Unable to merge because of this missing label. Thanks..

**Pull request closed without merging**